CellularBase/AT_CellularBase removal

AnttiKauppila opened the pull request:
Removed CellularBase and AT_CellularBase from cellular stack and updated both code and unittests accordingly.

<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).
-->

### Description 
#### Summary of change <!-- Required -->
Removed CellularBase and AT_CellularBase from cellular stack and updated both code and unittests accordingly. Also cellular drivers in Mbed OS have been updated.
Earlier cellular device properties were handled in AT_CellularBase class and after this change they are handled now correctly in AT_CellularDevice class. The properties are always modem specific and must be therefore handled in AT_CellularDevice class which is common for all contexts using it.

Also APN lookup and SMS features are now disabled by default
Disabling SMS will save ~4,5kB and APN lookup about ~2kB

<!-- Basic information about the change: what the change is for, why do we need it any implications -->

#### Documentation <!-- Optional, but most likely you need it -->

<!-- Details of any document updates required, including links to PR against the docs repository -->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [X] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Required
    Provide all the information required, listing all the testing performed. For new targets please attach full test results
    for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [X] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Optional
    Request additional reviewers with @username or @team
-->
@ARMmbed/mbed-os-wan 
----------------------------------------------------------------------------------------------------------------
### Release Notes <!-- Required for features, deprecations, breaking changes and other major PRs -->

<!--
    All 3 sections are compulsory for Major PR types. For Feature PRs only the summary section is required.
    This section is automatically added to release notes. Please fill in each sub-section with sufficient detail for a user.
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types). 
-->

#### Summary of changes

Removed CellularBase and AT_CellularBase from cellular stack and updated both code and unittests accordingly. Cellular drivers in Mbed OS have been updated. 

APN lookup and SMS features are now disabled by default

#### Impact of changes

Earlier cellular device properties were handled in AT_CellularBase class and after this change they are handled now correctly in AT_CellularDevice class. The properties are always modem specific and must be therefore handled in AT_CellularDevice class which is common for all contexts using it.

CellularBase has been completely removed so those using must now switch to use CellularInterface instead.
Disabling SMS will save ~4,5kB and APN lookup about ~2kB

#### Migration actions required
Cellular modem drivers must be updated (done for targets in Mbed OS) by removing references to CellularBase.
CellularBase must be changed to CellularInterface if still in use. (was already typedef'ed so no other changes needed)
If SMS or APN lookup are needed by application, those needs to be enabled in mbed_app.json
(cellular.use-apn-lookup: true or cellular.use-sms: true)

Commit:
Remove CellularBase and AT_CellularBase

Removed CellularBase and AT_CellularBase from cellular stack and updated both code and unittests accordingly.

Moved property handling into AT_CellularDevice

Commit:
Disable APN lookup by default

Commit:
CellularStack fixed

Commit:
Disabled SMS by default

kivaisan commented on features/cellular/framework/AT/AT_CellularDevice.h:
```diff
@@ -165,6 +206,9 @@ class AT_CellularDevice : public CellularDevice {
     void urc_pdn_deact();
+
+private:
+    const intptr_t *_property_array;
```
`_property_array` is not initialized in `AT_CellularDevice `constructor causing "TestAT_CellularDevice, Create" UT test to fail.

AnttiKauppila replied:
Fixed now

Commit:
Fixed unittests

Commit:
initialisation added for _property_array

Commit:
APN lookup fix

tools folder had a reference to ppp-cell-iface.apn-lookup which is removed.
This commit removes the reference to non-existing configuration
